ruby_event_store/lib/ruby_event_store/serialized_record.rb

```diff
@@ -0,0 +1,13 @@
+module RubyEventStore
+  class SerializedRecord
+    def initialize(id:, data:, metadata:, event_type:)
+      @id         = id
+      @data       = data
+      @metadata   = metadata
+      @event_type = event_type
+      freeze
+    end
```

Comment:
Do you think we should verify the interface that `id, data, metadata, event_type` are strings?

Reply:
That could possibly help avoid some errors.

Reply:
done